torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -1026,4 +1026,64 @@ xla::XlaOp BuildAddcmul(xla::XlaOp input, xla::XlaOp t1, xla::XlaOp t2,
       input, XlaHelpers::PromotedMul(XlaHelpers::PromotedMul(t1, t2), val));
 }
 
+xla::XlaOp BuildCdistForward(xla::XlaOp x1, xla::XlaOp x2, xla::XlaOp p,
+                             bool use_hamming, bool use_chebyshev) {
+  const xla::Shape& x1_shape = XlaHelpers::ShapeOfXlaOp(x1);
+  const xla::Shape& x2_shape = XlaHelpers::ShapeOfXlaOp(x2);
+  p = MaybeConvertTo(p, x1_shape.element_type());
+
+  XLA_CHECK(x1_shape.rank() == x2_shape.rank() && x1_shape.rank() >= 2)
+      << "x1 and x2 must have the same rank with >= 2 dimensions";
+
+  int rank = x1_shape.rank();
+
+  XLA_CHECK(x1_shape.dimensions(rank - 1) == x2_shape.dimensions(rank - 1))
+      << "The last dimension of x1 and x2 must match";
+
+  for (int dim = 0; dim < rank - 2; dim++) {
+    XLA_CHECK(x1_shape.dimensions(dim) == x2_shape.dimensions(dim))
+        << absl::StrCat("The ", dim, "th dimension of x1 and x2 must match");
+  }
+
+  std::vector<int64_t> bcast_shape(x1_shape.dimensions().begin(),
+                                   x1_shape.dimensions().end());
+  bcast_shape.insert(bcast_shape.begin() + rank - 1,
```

Comment:
hmm, are you trying to replace the last element of `bcast_shape` with `x2_shape.dimensions(rank - 2)`?

Reply:
It inserts `x2_shape.dimensions(rank - 2)` to `bcast_shape` so that `bcast_shape.dimensions(rank - 1) ==  x2_shape.dimensions(rank - 2)`.
For example, if we have `x1.shape == [b, m, k]` and `x2.shape == [b, n, k]`, then `bcast_shape == [b, m, n, k]`. This algorithm will broadcast `x1` and `x2` to `bcast_shape`, take the diff and then perform a reduce on the last dimension to get their pairwise p norms.